### ISSUE TITLE
Unable to open drawing shared via OneDrive for Business

### ISSUE THREAD

**Issue by Zxurian** [jgraph/drawio#3758]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
Sharing a drawing within OneDrive for Business to another user does not allow that user to open the shared drawing.

**To Reproduce**
1. Within drawing, click "Share" button in top right.
![image](https://github.com/jgraph/drawio/assets/573371/7ccc3f4f-18d1-42db-8380-3959a6490e36)
2. This redirects to a OneDrive file placeholder url.
3. Click Share button within OneDrive
![image](https://github.com/jgraph/drawio/assets/573371/9033f13e-7599-4c99-99d7-1276d57a10bc)
4. Enter user to share with
5. User 2 gets normal "<User 1> has shared a file with you"
6. User 2 clicks "Open" within share email
7. They are brought to the same OneDrive file placeholder url.
8. Open new tab and go directly to https://app.diagrams.net
9. Select "Open Existing Diagram"
10. Select "Shared" option. The file that User 1 shared with User 2 is shown as available
![image](https://github.com/jgraph/drawio/assets/573371/d2e0847e-3f1e-40b7-b03a-c1e668d0884c)
11. Click "Open"
12. Get error that file was not found.
![image](https://github.com/jgraph/drawio/assets/573371/5e1d50e0-285c-49ba-bc1f-abaabfa28074)

**Expected behavior**
Should be able to open the shared drawing

**Screenshots**
inline above with steps

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io version 21.6.6 for both users

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - Version: 115.0.5790.110

**Smartphone (please complete the following information):**
N/A

**Additional context**
Searching beforehand showed a possible link to #3755 however it's not quite the same since all the URLs work, the drawing just can't be opened.

**Comment by DavidPIsaac** [comment]
Same issue.  I have tried every conceivable path, including exporting and opening the URL (file > export as > url; file > open from > url).  The best I can achieve is the user can open a read-only copy even though they have read-write access to the file in OneDrive.  I expected the shared folder/file to be visible from the open dialog (file > open from > OneDrive > shared).

**Comment by cveld** [comment]
Not sure if my scenario applies to this issue:

I am added as guest to another tenant. With drawio I am only able to access the files in my primary tenant. Ideally I would like to be able to work with drawio files on SharePoint anywhere, regardsless if it it my primary tenant, or I am added as guest.

**Comment by sune-ku** [comment]
I also encountered this issue with shared drawings, but have discovered that if you double click on the file name, rather than clicking "Open" it does work correctly and allows for collaboration.

Clicking "Open" while having the file selected results in the Error message shown in the post from @Zxurian

**Comment by Floriszz** [comment]
For some reason I don't have this problem in the Edge browser with my primary profile. But when I open incognito and/or Chrome, I get the same problem. The thumbnail of the .draw.io file is not shown either. Also it doesn not matter if the file is in my Onedrive or in SharePoint.
Double click as @sune-ku mentions, does not work for me. Nothing happens after double click.
I tried with click on the ' log out'  button on the popup where you can choose to Open a file from Onedrive or create a new file. After signing out, you have to sign back in with the new Multifactor authentication where you have to type in the 2 digits on your mobile device(Microsoft Authenticator app). It looks like the problems started after this new authentication was introduced.
![image](https://github.com/jgraph/drawio/assets/19168937/56de8b57-bbe9-4683-8a90-ae5917839fa8)
![image](https://github.com/jgraph/drawio/assets/19168937/04b9bd04-be56-467a-b401-0780bbf57574)

**Comment by davidjgraph** [comment]
What are the draw.io versions in use in chrome and edge?

**Comment by Floriszz** [comment]
@davidjgraph, it is from app.diagrams.net website in the browser. I don't see where the user can choose a specific version.
Click on help shows v22.0.2.

**Comment by Floriszz** [comment]
Today, I tested on different internet connections(5G-Mobile, Wifi guest, LAN Home, LAN corporate, Wifi Enterprise Radius Windows login) and different Draw.IO apps; Edge, Chrome, Edge PWA(Progressive Web App).
And all worked fine! All draw.io files on different Onedrive Business locations could be opened.

**Comment by Floriszz** [comment]
F12 screenshot if of any help;
![image](https://github.com/jgraph/drawio/assets/19168937/3405840e-bfbf-4d1f-8efd-0592c3fc8e1c)

**Comment by davidjgraph** [comment]
Thanks for the additional research. It sounds like there's some restriction on your O365 org account. Do you know who the admin is to ask them?

**Comment by Zxurian** [comment]
I am an admin on our O365 org account, issue is still happening.
![image](https://github.com/jgraph/drawio/assets/573371/5959bfce-722c-4486-aead-24f4141966a8)

* If I go directly to draw.io, I can Authorize OneDrive for Business and open the doc from the list.
* If I hit "Share" in the top right, I get the same image at the top, however I can still share the document using M365 OneDrive sharing workflow
* When the new user gets the sharing link, if they click on it, they are presented with the same "file doesn't have a preview" image. If they click "Open", then it just refreshes the page with only a Download button.

**Comment by Floriszz** [comment]
The trick is to use the right click on the page tab and select link. Then configure the type of link you want to share(view/edit and/or 1 or multi pages)
Also to make it workable from Teams Files or SharePoint I alway add a new link file to SharePoint and point this link to either the vie or edit link mentioned above.
Sharing links from onedrive / SharePoint or Teams doesn't work. (Always results in download file)

**Comment by Floriszz** [comment]
> Thanks for the additional research. It sounds like there's some restriction on your O365 org account. Do you know who the admin is to ask them?

Yes, we are trying to find out which policy is the one. Will update when found

**Comment by Zxurian** [comment]
Revisiting this as it's still an issue.

Per #4228 I understand that you can't open the file directly from OneDrive, however from within draw.io The file still can't be found to open anywhere.
however the ability to open shared diagrams still does not work. the symptoms are the same as described previously above. https://github.com/jgraph/drawio/issues/3758#issuecomment-1749511857

I was able to take the direct url of the open document ( `https://app.diagrams.net/#...` ) and paste directly into the URL bar of the new user while they were signed into Office with that account, and it opened fine. However the existing issue of not being able to Open the shared document from draw.io's own interface still remained, and remains a high amount of user friction when trying to share draw.io documents between users.

**Comment by teohsinyee** [comment]
> I also encountered this issue with shared drawings, but have discovered that if you double click on the file name, rather than clicking "Open" it does work correctly and allows for collaboration.
> 
> Clicking "Open" while having the file selected results in the Error message shown in the post from @Zxurian

I am still encountering the issue. But double click on the file name worked.

**Comment by tokenflow** [comment]
Might look weird , as @teohsinyee mentioned just double click instead of Open and it will work

**Comment by Carlo-Piccini** [comment]
I get a 404 NOT FOUND when I click on the share button. I have tried every suggestion. My files are on OneDrive for business.

**Comment by cveld** [comment]
@davidjgraph would be fantastic to be able to easily switch between m365 accounts and tenants. Is this something on the roadmap? Can I contribute somehow? Currently I can only open files from my primary tenant. Would be also cool to be able to paste deeplinks to onedrive hosted drawio files.

**Comment by davidjgraph** [comment]
Try using the MS365 storage option in 29.6.0.

**Comment by davidjgraph** [comment]
This thread is quite long and probably out dated with the M365 introduction. I'd suggest, if you still have problems while using M365, to open new issues.
